static/app/views/performance/browser/webVitals/components/recommendations.tsx

```diff
@@ -0,0 +1,169 @@
+
+export function Recommendations({
+  transaction,
+  webVital,
+}: {
+  transaction: string;
+  webVital: WebVitals;
+}) {
+  switch (webVital) {
+    case 'lcp':
+      return null;
+    case 'fid':
+      return null;
+    case 'cls':
+      return null;
+    case 'fcp':
+      return <FcpRecommendations transaction={transaction} />;
+    case 'ttfb':
+      return null;
+    default:
+      return null;
+  }
+}
+
+function FcpRecommendations({transaction}: {transaction: string}) {
+  const query = `transaction:"${transaction}" resource.render_blocking_status:blocking`;
+  const {data, isLoading} = useResourcesQuery({
+    query,
+    sort: {field: `avg(${SpanMetricsField.SPAN_SELF_TIME})`, kind: 'desc'},
+    defaultResourceTypes: ['resource.script', 'resource.css', 'resource.img'],
+    limit: 7,
+  });
+  if (isLoading || !data) {
+    return null;
+  }
+  return (
+    <RecommendationsContainer>
+      <RecommendationsHeader />
+      <ul>
+        <RecommendationSubHeader>
+          {t('Eliminate render blocking resources')}
+        </RecommendationSubHeader>
+        <ResourceList>
+          {data.map(
+            ({
+              'span.op': op,
+              'span.description': description,
+              'avg(span.self_time)': duration,
+            }) => {
+              return (
+                <ResourceListItem key={description}>
+                  <Flex>
+                    <span>
+                      <ResourceType resourceType={op} />
+                      <ResourceDescription>
+                        {formatDescription(description)}
+                      </ResourceDescription>
+                    </span>
+                    <span>{getFormattedDuration(duration)}</span>
+                  </Flex>
+                </ResourceListItem>
+              );
+            }
+          )}
+        </ResourceList>
+      </ul>
+    </RecommendationsContainer>
+  );
+}
+
+function RecommendationsHeader() {
+  return (
+    <RecommendationsHeaderContainer>
+      <b>{t('Recommendations')}</b>
+    </RecommendationsHeaderContainer>
+  );
+}
+
+function ResourceType({resourceType}: {resourceType: `resource.${string}`}) {
+  switch (resourceType) {
+    case 'resource.script':
+      return (
+        <b>
+          <StyledIconFile size="xs" />
+          {t('js')}
+          {' \u2014 '}
+        </b>
+      );
+    case 'resource.css':
+      return (
+        <b>
+          <StyledIconFile size="xs" />
+          {t('css')}
+          {' \u2014 '}
+        </b>
+      );
+    case 'resource.img':
+      return (
+        <b>
+          <StyledIconFile size="xs" />
+          {t('img')}
+          {' \u2014 '}
+        </b>
+      );
+    default:
+      return null;
+  }
+}
+
+function formatDescription(description: string) {
+  // Center truncate the description
+  if (description.length > 50) {
+    return `${description.slice(0, 25)}...${description.slice(description.length - 25)}`;
```

Comment:
imo it's confusing to me that we truncate the middle of the string. Can we truncate the end and make it so that you can hover the description to see it in full?

Reply:
Yeah I think this makes more sense and follows existing patterns, updated!
(Had to tweak css a bit to fit the tooltip component in)
![image](https://github.com/getsentry/sentry/assets/83961295/b6814e9e-15b3-428a-b61b-660c47948f12)